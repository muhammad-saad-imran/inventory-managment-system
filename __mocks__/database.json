{
  "clients": [
    {
      "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
      "name": "Alice Smith",
      "email": "alice.smith@example.com",
      "phone_number": 1234567890,
      "created_at": "2024-01-01T12:00:00Z"
    },
    {
      "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
      "name": "Bob Johnson",
      "email": "bob.johnson@example.com",
      "phone_number": 9876543210,
      "created_at": "2024-01-05T15:30:00Z"
    }
  ],
  "products": [
    {
      "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
      "name": "Product A",
      "description": "Description of Product A",
      "created_at": "2024-02-01T09:00:00Z"
    },
    {
      "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
      "name": "Product B",
      "description": "Description of Product B",
      "created_at": "2024-02-15T10:00:00Z"
    }
  ],
  "suppliers": [
    {
      "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
      "name": "Supplier One",
      "email": "supplier.one@example.com",
      "phone_number": 5551234567,
      "created_at": "2024-03-01T08:00:00Z"
    },
    {
      "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
      "name": "Supplier Two",
      "email": "supplier.two@example.com",
      "phone_number": 5559876543,
      "created_at": "2024-03-10T14:00:00Z"
    }
  ],
  "inventory": [
    {
      "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
      "cost": 50.00,
      "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
      "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
      "stock_quantity": 100,
      "reorder_limit": 20,
      "supply_date": "2024-03-15T10:00:00Z",
      "created_at": "2024-03-15T10:00:00Z",
      "suppliers": {
        "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
        "name": "Supplier One",
        "email": "supplier.one@example.com",
        "phone_number": 5551234567,
        "created_at": "2024-03-01T08:00:00Z"
      },
      "products": {
        "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
        "name": "Product A",
        "description": "Description of Product A",
        "created_at": "2024-02-01T09:00:00Z"
      }
    },
    {
      "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
      "cost": 75.00,
      "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
      "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
      "stock_quantity": 50,
      "reorder_limit": 10,
      "supply_date": "2024-03-20T11:00:00Z",
      "created_at": "2024-03-20T11:00:00Z",
      "suppliers": {
        "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
        "name": "Supplier Two",
        "email": "supplier.two@example.com",
        "phone_number": 5559876543,
        "created_at": "2024-03-10T14:00:00Z"
      },
      "products": {
        "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
        "name": "Product B",
        "description": "Description of Product B",
        "created_at": "2024-02-15T10:00:00Z"
      }
    }
  ],
  "orders": [
    {
      "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
      "client_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
      "order_date": "2024-04-01T12:00:00Z",
      "status": "PROCESSING",
      "created_at": "2024-04-01T12:00:00Z",
      "clients": {
        "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
        "name": "Alice Smith",
        "email": "alice.smith@example.com",
        "phone_number": 1234567890,
        "created_at": "2024-01-01T12:00:00Z"
      },
      "order_items": [
        {
          "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b1",
          "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
          "inventory_id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
          "quantity": 2,
          "price": 100.00,
          "created_at": "2024-04-01T12:00:00Z",
          "inventory": {
            "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
            "cost": 50.00,
            "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
            "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
            "stock_quantity": 100,
            "reorder_limit": 20,
            "supply_date": "2024-03-15T10:00:00Z",
            "created_at": "2024-03-15T10:00:00Z",
            "suppliers": {
              "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
              "name": "Supplier One",
              "email": "supplier.one@example.com",
              "phone_number": 5551234567,
              "created_at": "2024-03-01T08:00:00Z"
            },
            "products": {
              "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
              "name": "Product A",
              "description": "Description of Product A",
              "created_at": "2024-02-01T09:00:00Z"
            }
          }
        }
      ]
    },
    {
      "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
      "client_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
      "order_date": "2024-04-05T15:30:00Z",
      "status": "SHIPPED",
      "created_at": "2024-04-05T15:30:00Z",
      "clients": {
        "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
        "name": "Bob Johnson",
        "email": "bob.johnson@example.com",
        "phone_number": 9876543210,
        "created_at": "2024-01-05T15:30:00Z"
      },
      "order_items": [
        {
          "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b2",
          "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
          "inventory_id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
          "quantity": 1,
          "price": 75.00,
          "created_at": "2024-04-05T15:30:00Z",
          "inventory": {
            "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
            "cost": 75.00,
            "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
            "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
            "stock_quantity": 50,
            "reorder_limit": 10,
            "supply_date": "2024-03-20T11:00:00Z",
            "created_at": "2024-03-20T11:00:00Z",
            "suppliers": {
              "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
              "name": "Supplier Two",
              "email": "supplier.two@example.com",
              "phone_number": 5559876543,
              "created_at": "2024-03-10T14:00:00Z"
            },
            "products": {
              "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
              "name": "Product B",
              "description": "Description of Product B",
              "created_at": "2024-02-15T10:00:00Z"
            }
          }
        }
      ]
    }
  ],
  "invoices": [
    {
      "id": "c7f9d0d5-9a30-4a26-b0c4-00ff18b8b781",
      "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
      "amount": 200.00,
      "invoice_date": "2024-04-02T12:00:00Z",
      "order": {
        "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
        "client_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
        "order_date": "2024-04-01T12:00:00Z",
        "status": "PROCESSING",
        "created_at": "2024-04-01T12:00:00Z",
        "clients": {
          "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
          "name": "Alice Smith",
          "email": "alice.smith@example.com",
          "phone_number": 1234567890,
          "created_at": "2024-01-01T12:00:00Z"
        },
        "order_items": [
          {
            "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b1",
            "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
            "inventory_id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
            "quantity": 2,
            "price": 100.00,
            "created_at": "2024-04-01T12:00:00Z",
            "inventory": {
              "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
              "cost": 50.00,
              "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
              "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
              "stock_quantity": 100,
              "reorder_limit": 20,
              "supply_date": "2024-03-15T10:00:00Z",
              "created_at": "2024-03-15T10:00:00Z",
              "suppliers": {
                "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
                "name": "Supplier One",
                "email": "supplier.one@example.com",
                "phone_number": 5551234567,
                "created_at": "2024-03-01T08:00:00Z"
              },
              "products": {
                "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
                "name": "Product A",
                "description": "Description of Product A",
                "created_at": "2024-02-01T09:00:00Z"
              }
            }
          }
        ]
      }
    },
    {
      "id": "a9f5d0d4-90ff-4a69-b0b5-7b8e3c5fbb9d",
      "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
      "amount": 75.00,
      "invoice_date": "2024-04-06T15:30:00Z",
      "order": {
        "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
        "client_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
        "order_date": "2024-04-05T15:30:00Z",
        "status": "SHIPPED",
        "created_at": "2024-04-05T15:30:00Z",
        "clients": {
          "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
          "name": "Bob Johnson",
          "email": "bob.johnson@example.com",
          "phone_number": 9876543210,
          "created_at": "2024-01-05T15:30:00Z"
        },
        "order_items": [
          {
            "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b2",
            "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
            "inventory_id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
            "quantity": 1,
            "price": 75.00,
            "created_at": "2024-04-05T15:30:00Z",
            "inventory": {
              "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
              "cost": 75.00,
              "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
              "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
              "stock_quantity": 50,
              "reorder_limit": 10,
              "supply_date": "2024-03-20T11:00:00Z",
              "created_at": "2024-03-20T11:00:00Z",
              "suppliers": {
                "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
                "name": "Supplier Two",
                "email": "supplier.two@example.com",
                "phone_number": 5559876543,
                "created_at": "2024-03-10T14:00:00Z"
              },
              "products": {
                "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
                "name": "Product B",
                "description": "Description of Product B",
                "created_at": "2024-02-15T10:00:00Z"
              }
            }
          }
        ]
      }
    }
  ]
}
