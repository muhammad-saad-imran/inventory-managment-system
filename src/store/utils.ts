import { completeLoading, startLoading } from "@/store/features/loading";
import { createAsyncThunk, GetThunkAPI } from "@reduxjs/toolkit";

export const getThunkCreator = <T>(
  thunkName: string,
  getEntity: (id: string) => Promise<T>
) =>
  createAsyncThunk(
    thunkName,
    async (id: string, { dispatch, rejectWithValue }) => {
      try {
        dispatch(startLoading());
        const entity = await getEntity(id);
        dispatch(completeLoading());
        return entity;
      } catch (error) {
        dispatch(completeLoading());
        alert("Error fetching products");
        rejectWithValue(error);
      }
    }
  );

export const getAllThunkCreator = <T>(
  thunkName: string,
  getAllEntity: (input: string) => Promise<T[]>
) =>
  createAsyncThunk(
    thunkName,
    async (name: string, { dispatch, rejectWithValue }) => {
      try {
        dispatch(startLoading());
        const entity = await getAllEntity(name);
        dispatch(completeLoading());
        return entity;
      } catch (error) {
        dispatch(completeLoading());
        alert("Error fetching products");
        rejectWithValue(error);
      }
    }
  );

export const mutationThunkCreator = <T>(
  thunkName: string,
  mutateEntity: (values: T) => Promise<T>
) =>
  createAsyncThunk(
    thunkName,
    async (values: T, { dispatch, rejectWithValue }) => {
      try {
        dispatch(startLoading());
        const entity = await mutateEntity(values);
        dispatch(completeLoading());
        return entity;
      } catch (error) {
        dispatch(completeLoading());
        alert("Error fetching products");
        rejectWithValue(error);
      }
    }
  );

export const deleteThunkCreator = (
  thunkName: string,
  deleteEntity: (id: string) => Promise<void>
) =>
  createAsyncThunk(
    thunkName,
    async (id: string, { dispatch, rejectWithValue }) => {
      try {
        dispatch(startLoading());
        await deleteEntity(id);
        dispatch(completeLoading());
        return;
      } catch (error) {
        dispatch(completeLoading());
        alert("Error fetching products");
        rejectWithValue(error);
      }
    }
  );
