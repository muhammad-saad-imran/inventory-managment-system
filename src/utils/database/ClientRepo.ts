import { GenericRepo } from "@/utils/database/GenericRepo";
import { Client, DB_TABLES } from "@/utils/database/types";
import { SupabaseClient } from "@supabase/supabase-js";

export class ClientRepo extends GenericRepo<Client> {
  constructor(_client: SupabaseClient) {
    super(_client, DB_TABLES.CLIENTS);
  }

  async getWithName(name: string): Promise<Client[]> {
    const { data: entity, error } = await this.supabase
      .from(this.tableName)
      .select()
      .ilike("name", `%${name}%`);

    if (entity) {
      return entity;
    } else {
      throw error;
    }
  }
}
