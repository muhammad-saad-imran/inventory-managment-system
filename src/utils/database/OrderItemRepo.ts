import { SupabaseClient } from "@supabase/supabase-js";
import { GenericRepo } from "@/utils/database/GenericRepo";
import { DB_TABLES, OrderItem } from "@/utils/database/types";

export class OrderItemRepo extends GenericRepo<OrderItem> {
  constructor(_client: SupabaseClient) {
    super(_client, DB_TABLES.ORDER_ITEMS);
  }

  override async getAll(
    order_id?: string,
    include?: string
  ): Promise<OrderItem[]> {
    let query = this.supabase.from(this.tableName).select(include);

    if (order_id) {
      query.eq("order_id", order_id);
    }

    const { data, error } = await query;

    if (data) {
      return data as unknown as OrderItem[];
    } else {
      throw error;
    }
  }
}
