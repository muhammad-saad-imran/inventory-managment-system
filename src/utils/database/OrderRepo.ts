import { SupabaseClient } from "@supabase/supabase-js";
import { GenericRepo } from "@/utils/database/GenericRepo";
import { DB_TABLES, Order } from "@/utils/database/types";

export class OrderRepo extends GenericRepo<Order> {
  constructor(_client: SupabaseClient) {
    super(_client, DB_TABLES.ORDERS);
  }

  async getWithClientName(clientName: string) {
    const { data: allOrders, error } = await this.supabase
      .from(this.tableName)
      .select("*, clients!inner (*), order_items (*)")
      .ilike("clients.name", `%${clientName}%`)
      .order("order_date", { ascending: false });

    if (error) throw error;

    return allOrders;
  }
}
