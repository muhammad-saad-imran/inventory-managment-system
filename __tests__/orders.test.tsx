import { act } from "@testing-library/react";
import { fireEvent, screen, waitFor } from "@testing-library/dom";
import { renderWithProviders } from "@/utils/test-utils";
import { Order, ORDER_STATUS } from "@/utils/database/types";
import { ClientRepo } from "@/utils/database/ClientRepo";
import { OrderRepo } from "@/utils/database/OrderRepo";
import { OrderItemRepo } from "@/utils/database/OrderItemRepo";
import database from "@/../__mocks__/database.json";
import OrderPage from "@/app/dashboard/orders/page";
import OrderInfoPage from "@/app/dashboard/orders/[id]/page";
import selectEvent from "react-select-event";

jest.mock("next/navigation", () => {
  const actualImport = jest.requireActual("next/navigation");
  return {
    ...actualImport,
    usePathname() {
      return "/dashboard/orders";
    },
  };
});

jest.mock("@/utils/supabase/client");
jest.mock("@/utils/database/OrderRepo");
jest.mock("@/utils/database/ClientRepo");
jest.mock("@/utils/database/OrderItemRepo");
jest.mock("@/utils/database/InventoryRepo");

describe("Order feature", () => {
  // Order Repo mocks
  let getAllOrdersMock = jest.spyOn(OrderRepo.prototype, "getWithClientName");
  let getOrderMock = jest.spyOn(OrderRepo.prototype, "get");
  let createOrderMock = jest.spyOn(OrderRepo.prototype, "create");
  let updateOrderMock = jest.spyOn(OrderRepo.prototype, "update");

  // OrderItems service mocks
  let getAllOrderItemsMock = jest.spyOn(OrderItemRepo.prototype, "getAll");
  let getItemMock = jest.spyOn(OrderItemRepo.prototype, "get");
  let createOrderItemMock = jest.spyOn(OrderItemRepo.prototype, "create");
  let updateOrderItemMock = jest.spyOn(OrderItemRepo.prototype, "update");
  let deleteOrderItemMock = jest.spyOn(OrderItemRepo.prototype, "delete");

  // Cliets Repo mocks
  let getClientsMock = jest.spyOn(ClientRepo.prototype, "getWithName");

  let useRouter = jest.spyOn(require("next/navigation"), "useRouter");
  let useParams = jest.spyOn(require("next/navigation"), "useParams");
  const routerPushMock = jest.fn();

  beforeEach(() => {
    // mock useRouter implementation
    useRouter.mockImplementation(() => ({
      push: routerPushMock,
    }));

    // mock useParams implementation
    useParams.mockImplementation(() => ({ id: database.orders[0].id }));

    // mock Order Repo methods
    getOrderMock.mockResolvedValue(database.orders[0] as Order);
    getAllOrdersMock.mockResolvedValue(database.orders as Order[]);
    createOrderMock.mockResolvedValue(database.orders[0] as Order);
    updateOrderMock.mockResolvedValue(database.orders[0] as Order);

    // mock OrderItemService methods
    getAllOrderItemsMock.mockResolvedValue(database.orders[0].order_items);
    getItemMock.mockResolvedValue(database.orders[0].order_items[0]);
    createOrderItemMock.mockResolvedValue(database.orders[0].order_items[0]);
    updateOrderItemMock.mockResolvedValue(database.orders[0].order_items[0]);

    // mock Client Repo methods
    getClientsMock.mockResolvedValue(database.clients);
  });

  afterEach(() => {
    // Clean up mocks after each test
    useRouter.mockClear();
    useParams.mockClear();
    getOrderMock.mockClear();
    getAllOrdersMock.mockClear();
    createOrderMock.mockClear();
    updateOrderMock.mockClear();
    getAllOrderItemsMock.mockClear();
    getItemMock.mockClear();
    createOrderItemMock.mockClear();
    updateOrderItemMock.mockClear();
    getClientsMock.mockClear();
  });

  describe("OrderPage", () => {
    it("should match snapshot", () => {
      const { container } = renderWithProviders(<OrderPage />);
      expect(container).toMatchSnapshot();
    });

    it("should route on clicking order table row", async () => {
      renderWithProviders(<OrderPage />);

      const elems = await screen.findAllByTestId("order-row");
      fireEvent.click(elems[0]);

      expect(routerPushMock).toHaveBeenCalledTimes(1);
    });

    it("should create order with CreateOrderBar component", async () => {
      renderWithProviders(<OrderPage />);

      await screen.findByLabelText(/client/i);
      
      await selectEvent.select(
        await screen.findByLabelText(/client/i),
        /alice smith/i
      );

      await fireEvent.change(await screen.findByLabelText(/Order Date/i), {
        target: { value: "2025-12-12" },
      });

      await fireEvent.click(
        await screen.findByRole("button", { name: /create/i })
      );

      await waitFor(() => expect(createOrderMock).toHaveBeenCalled());
    });
  });

  describe("OrderInfoPage", () => {
    it("should match snapshot", () => {
      const { container } = renderWithProviders(<OrderInfoPage />);
      expect(container).toMatchSnapshot();
    });

    it("should update order status", async () => {
      await act(() => renderWithProviders(<OrderInfoPage />));

      await act(async () => {
        fireEvent.change(await screen.findByLabelText(/Order Status/i), {
          target: { value: ORDER_STATUS.SHIPPED },
        });

        fireEvent.click(await screen.findByRole("button", { name: /update/i }));
      });

      expect(updateOrderMock).toHaveBeenCalled();
    });

    it("should increment orderItem", async () => {
      await act(() => renderWithProviders(<OrderInfoPage />));

      await act(async () => {
        fireEvent.click(await screen.getByText(/\+/i));
      });

      expect(updateOrderItemMock).toHaveBeenCalled();
    });

    it("should increment orderItem", async () => {
      await act(() => renderWithProviders(<OrderInfoPage />));

      await act(async () => {
        fireEvent.click(await screen.getByText(/\-/i));
      });

      expect(updateOrderItemMock).toHaveBeenCalled();
    });

    it("should delete orderItem", async () => {
      await act(() => renderWithProviders(<OrderInfoPage />));

      await act(async () => {
        fireEvent.click((await screen.findAllByTestId("delete-btn"))[0]);
      });

      expect(deleteOrderItemMock).toHaveBeenCalled();
    });
  });
});
