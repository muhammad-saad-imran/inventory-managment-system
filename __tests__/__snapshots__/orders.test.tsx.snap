// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order feature OrderInfoPage should match snapshot 1`] = `
<div>
  <div>
    <p
      class="text-3xl text-center"
    >
      Order
    </p>
    <div
      class="flex flex-col gap-3 bg-white w-full p-5 mt-3"
    >
      <div
        class="flex gap-3"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label>
            Client
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            disabled=""
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label>
            Order date
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            disabled=""
            value="Invalid Date"
          />
        </div>
      </div>
      <div
        class="flex gap-4 w-full"
      >
        <div
          class="w-full"
        >
          <label
            for="order-status"
          >
            Order Status
          </label>
          <select
            class="sc-23ee340d-2 sc-23ee340d-3 jBqjoF jWMivj mt-2"
            id="order-status"
            name="order-status"
          >
            <option
              value="PROCESSING"
            >
              Processing
            </option>
            <option
              value="SHIPPED"
            >
              Shipped
            </option>
            <option
              value="DELIVERED"
            >
              Delivered
            </option>
            <option
              value="CANCELLED"
            >
              Cancelled
            </option>
          </select>
        </div>
        <button
          class="sc-7d679c72-0 sc-7d679c72-1 gkXiWL iTQhfb h-9 self-end"
          disabled=""
        >
          Update
        </button>
      </div>
    </div>
    <form
      class="bg-white w-full p-5 mt-3"
    >
      <div
        class="flex gap-4 w-full"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="Inventory"
          >
            Inventory
          </label>
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-5-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-hlgwow"
              >
                <div
                  class=" css-1jqq78o-placeholder"
                  id="react-select-5-placeholder"
                >
                  Select...
                </div>
                <div
                  class=" css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-5-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="Inventory"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  class=" css-pogzpp-loadingIndicator"
                >
                  <span
                    class="css-1h1juvu-LoadingDot"
                  />
                  <span
                    class="css-1swgd6a-LoadingDot"
                  />
                  <span
                    class="css-41hts2-LoadingDot"
                  />
                </div>
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="Inventory"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label>
            Stock
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            disabled=""
            value="0"
          />
        </div>
      </div>
      <div
        class="flex gap-4 w-full mt-3"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label>
            Cost
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            disabled=""
            value="$0.00"
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="price"
          >
            Selling Price
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="price"
            id="price"
            name="price"
            step="0.01"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="flex gap-4 w-full mt-3"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="quantity"
          >
            Quantity
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="quantity"
            id="quantity"
            name="quantity"
            type="number"
            value="0"
          />
        </div>
        <div
          class="w-full flex gap-4"
        >
          <div
            class="flex flex-col gap-1 w-full"
          >
            <label>
              Total Price
            </label>
            <input
              class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
              disabled=""
              value="$0.00"
            />
          </div>
          <button
            class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb h-9 self-end"
            type="submit"
          >
            Add
          </button>
        </div>
      </div>
    </form>
    <table
      class="w-full bg-white mt-3 p-3 text-center"
    >
      <thead>
        <tr>
          <th
            class="py-5 w-1/3"
          >
            Product
          </th>
          <th
            class="py-5 w-1/3"
          >
            Quantity
          </th>
          <th
            class="py-5 w-1/3"
          >
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="border-t border-b"
        >
          <td
            class="py-5 w-1/3"
          />
          <td
            class="py-5 w-1/3"
          />
          <td
            class="py-5 w-1/3 text-lg"
          >
            Total 
            <span
              class="text-xl"
            >
              $0.00
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Order feature OrderPage should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-3"
  >
    <div>
      <p
        class="text-3xl text-center"
      >
        Orders
      </p>
      <div
        class="flex gap-8 bg-white w-full p-5 mt-3"
      >
        <input
          class="sc-23ee340d-0 cwYaeo border-0 text-lg border-b"
          placeholder="Search"
          value=""
        />
      </div>
    </div>
    <form
      class="flex gap-4 bg-white w-full p-5"
    >
      <div
        class="flex flex-col gap-1 w-full"
      >
        <label
          for="Client"
        >
          Client
        </label>
        <div
          class=" css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
            role="log"
          />
          <div
            class=" css-13cymwt-control"
          >
            <div
              class=" css-hlgwow"
            >
              <div
                class=" css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              >
                Select...
              </div>
              <div
                class=" css-19bb58m"
                data-value=""
              >
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="Client"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1wy0on6"
            >
              <div
                aria-hidden="true"
                class=" css-pogzpp-loadingIndicator"
              >
                <span
                  class="css-1h1juvu-LoadingDot"
                />
                <span
                  class="css-1swgd6a-LoadingDot"
                />
                <span
                  class="css-41hts2-LoadingDot"
                />
              </div>
              <span
                class=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="Client"
            type="hidden"
            value=""
          />
        </div>
      </div>
      <div
        class="flex flex-col gap-1 w-full"
      >
        <label
          for="order_date"
        >
          Order Date
        </label>
        <input
          class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
          data-testid="order_date"
          id="order_date"
          name="order_date"
          type="date"
          value=""
        />
      </div>
      <button
        class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb h-9 self-end"
      >
        Create
      </button>
    </form>
    <table
      class="w-full bg-white"
    >
      <thead>
        <tr>
          <th
            class="py-5 w-1/4"
          >
            ID
          </th>
          <th
            class="py-5 w-1/4"
          >
            Client
          </th>
          <th
            class="py-5 w-1/4"
          >
            Date
          </th>
          <th
            class="py-5 w-1/4"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
